src/librustc_mir/interpret/error.rs

```diff
@@ -11,7 +13,41 @@ use rustc_const_math::ConstMathErr;
 use syntax::codemap::Span;
 
 #[derive(Debug)]
-pub enum EvalError<'tcx> {
+pub struct EvalError<'tcx> {
+    pub kind: EvalErrorKind<'tcx>,
+    pub backtrace: Vec<Frame>,
+}
+
+impl<'tcx> From<EvalErrorKind<'tcx>> for EvalError<'tcx> {
+    fn from(kind: EvalErrorKind<'tcx>) -> Self {
+        let mut backtrace = Vec::new();
+        use backtrace::{trace, resolve};
+        trace(|frame| {
+            resolve(frame.ip(), |symbol| {
+                backtrace.push(Frame {
+                    function: symbol.name().map(|s| s.to_string()).unwrap_or(String::new()),
+                    file: symbol.filename().unwrap_or(Path::new("")).to_owned(),
+                    line: symbol.lineno().unwrap_or(0),
+                });
+            });
+            true
+        });
```

Comment:
The backtrace crate also has this type which just captures the entire backtrace at once:
<http://alexcrichton.com/backtrace-rs/backtrace/struct.Backtrace.html>
Why did you end up doing the more manual capturing here?